azurerm_virtual_network_gateway_connection  - add "connection_protocol" property (#5145)
terraform-providers/terraform-provider-azurerm: website/docs/r/virtual_network_gateway_connection.html.markdown

    Site-to-Site or VNet-to-VNet connection is created whereas ExpressRoute
    connections do not need a shared key.

* `connection_protocol` - (Optional) The IKE protocol version to use. Possible
    values are `IKEv1` and `IKEv2`. Only valid for IPSec connections on virtual
    network gateways with SKU `VpnGw1`, `VpnGw2`, `VpnGw3`, `VpnGw1AZ`, `VpnGw2AZ`

>     or `VpnGw3AZ`. Changing the protocol will force a new connection to be created. 

Reviewer: could we split this line:

```
Only valid for IPSec connections on virtual network gateways with SKU `VpnGw1`, `VpnGw2`, `VpnGw3`, `VpnGw1AZ`, `VpnGw2AZ` or `VpnGw3AZ`.
```

into a blue info box below the description? We can do this using:

```
-> Only valid for IPSec connections on Virtual Network Gateways using the SKU `VpnGw1`, `VpnGw2`, `VpnGw3`, `VpnGw1AZ`, `VpnGw2AZ` or `VpnGw3AZ`.
```

in addition could we use the text:

```suggestion
Changing this value forces a new resource to be created.
```

to match the rest of the documentation?

>     Defaults to `IKEv2`. 

Reviewer: There's no default set above? As such should we set this to:

```suggestion
```

Author: It is implicit set to IKEv2 by the API if the value is emtpy. Hence the `if` check above

Author: Also I wouldn't want to introduce it as a default value on the provider, as it would cause all existing connections to be recreated. But as the default if not set is IKEv2 I have no better idea on how to document this...
